lib/data/wrapper.js

```diff
@@ -26,6 +26,22 @@ if (config.backends.data === 'mem') {
     implName = 'sproxyd';
 }
 
+/**
+ * _retryDelete - Attempt to delete key again if it failed previously
+ * @param  {string} key - location of the object to delete
+ * @param  {object} log - Werelogs request logger
+ * @returns {boolean} true or false - indicates whether delete was successful
+ */
+function _retryDelete(key, log) {
```

Comment:
Hmm.. yeah. So you're saying because the call to client.delete() is asynch, _retryDelete will finish execution and return before client.delete() executes its callback, right? Thanks for the catch, I'm still trying to get a good handle on this topic. Your recursive solution looks good, I will update with the changes and test my previous code to make sure I understand.